File: temp_pdf_parser/extract.py
import PyPDF2

def extract_text_from_pdf(pdf_path, txt_path):
    with open(pdf_path, 'rb') as file:
        reader = PyPDF2.PdfReader(file)
        text = ''
        for page_num in range(len(reader.pages)):
            page = reader.pages[page_num]
            text += f'--- Page {page_num + 1} ---\n'
            text += page.extract_text() + '\n\n'
    
    with open(txt_path, 'w', encoding='utf-8') as file:
        file.write(text)

if __name__ == '__main__':
    extract_text_from_pdf('../WAREHOUSE MANAGEMENT SYSTEM.pdf', 'extracted_wms.txt')
    print('Successfully extracted PDF text to extracted_wms.txt')

File: temp_pdf_parser/extract.js
const fs = require('fs');
const pdf = require('pdf-parse');

const dataBuffer = fs.readFileSync('../WAREHOUSE MANAGEMENT SYSTEM.pdf');

pdf(dataBuffer).then(function(data) {
    fs.writeFileSync('extracted_wms.txt', data.text);
    console.log('PDF extracted successfully to extracted_wms.txt');
}).catch(err => {
    console.error('Error parsing PDF:', err);
});
